game/hud/src/components/AnchorIndicator/index.tsx

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import {Position} from '../../services/session/layout';
+
+export interface AnchorIndicatorProps {
+  name: string;     // widget name
+  pos: Position;    // current widget position
+  setAnchor: (name: string, pos: Position) => void;
+}
+
+export interface AnchorIndicatorState {
+}
+
+class AnchorIndicator extends React.Component<AnchorIndicatorProps, AnchorIndicatorState> {
+
+  constructor(props: AnchorIndicatorProps) {
+    super(props);
+  }
+
+  private anchorHere = (event: React.MouseEvent) => {
+    const el: HTMLDivElement = event.currentTarget as HTMLDivElement;
+    function a2n(a: string) : number {
+      switch(a) {
+        case 's': return -1;
+        case 'e': return 1;
+      }
+      return 0;
+    }
+    const pos: Position = Object.assign({}, this.props.pos);
+    let x: number = a2n(el.classList[1][7]);
+    let y: number = a2n(el.classList[2][7]);
+    if (x === pos.anchor.x && y === pos.anchor.y) {
+      pos.anchor.locked = !pos.anchor.locked;
+    } else {
+      pos.anchor.locked = true;
+      pos.anchor.x = x;
+      pos.anchor.y = y;
+    }
+    this.props.setAnchor(this.props.name, pos);
+  }
+
+  render() {
+    const anchors: JSX.Element[] = [];
+    const pos: Position = this.props.pos;
```

Comment:
Props won't, pos might be, this would probably be a good place to throw or exit if you weren't handed a good pos. I don't think you'd want to run any of the following code if you didn't have a good pos?